main.go

```diff
@@ -201,9 +211,35 @@ func LoadBundles (wh warehouse.Warehouse, filename string, bundles ...fullstory.
 	return nil
 }
 
+func getExportData(fs *fullstory.Client, bundleID int) (fullstory.ExportData, error) {
+	log.Printf("Getting Export Data for bundle %d\n", bundleID)
+	for r := 1; r <= maxAttempts; r++ {
+		stream, err := fs.ExportData(bundleID)
+		if err == nil {
+			return stream, nil
+		}
+
+		log.Printf("Failed to fetch export data for Bundle %d", bundleID)
+		if statusError, ok := err.(fullstory.StatusError); ok {
+			// If the status code is NOT 429 and the code is below 500 we will not attempt to retry
+			if statusError.StatusCode != http.StatusTooManyRequests && statusError.StatusCode < 500 {
+				break
+			}
+
+			retryAfterDuration := defaultRetryAfterDuration
+			if retryAfter, err := strconv.Atoi(statusError.RetryAfter); err == nil {
+				retryAfterDuration = time.Duration(retryAfter) * time.Second
+			}
+			log.Printf("Attempt #%d failed. Retrying after %s\n", r, retryAfterDuration)
+			time.Sleep(retryAfterDuration)
+		}
```

Comment:
I guess we intend to retry if `ok` is false, i.e. `err` is not a `fullstory.StatusError`? I think such an error might occur for example if the request never made it to the FS server due to temporary network issues, so it's correct to retry it. 

Right now if I'm reading this right, the retry does occur, but right away. Maybe it's better to put the init logic for `retryAfterDuration` and for the call to `time.Sleep` outside of the check to see whether this is a statusError, so we always retry on early failures, but after the default wait.

Reply:
good point. done. Also any thoughts on the 10s default delay between retries, now that we are re-trying a larger set of errors?

Reply:
I think 10s is fine. Worst case scenario, it fails after 3 tries and then bubbles the error up to the main algorithm, which waits 30s before trying again after errors. Effectively this just means some errors that occurred while fetching bundles will be retried a bit faster (10s vs 30s).